core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraBuilder.kt

```diff
@@ -259,15 +258,8 @@ interface RestrictedRawInfraBuilder {
     fun linkZones(detector: DetectorId, zoneA: ZoneId, zoneB: ZoneId)
 
     fun setNextZone(detector: DirDetectorId, zone: ZoneId)
```

Comment:
as described below, the zone method wasn't designed to create a complete zone from the get go: you create a zone handle, and then use setNextZone to define its bounds. I believe the builder API can be simplified to make the job simpler

Reply:
See the conversation https://github.com/osrd-project/osrd/pull/6921#discussion_r1543018849